main/src/cgeo/geocaching/filters/gui/DifficultyAndTerrainFilterViewHolder.java

```diff
@@ -0,0 +1,67 @@
+package cgeo.geocaching.filters.gui;
+
+import cgeo.geocaching.R;
+import cgeo.geocaching.filters.core.DifficultyAndTerrainGeocacheFilter;
+import cgeo.geocaching.filters.core.DifficultyGeocacheFilter;
+import cgeo.geocaching.filters.core.GeocacheFilterType;
+import cgeo.geocaching.filters.core.TerrainGeocacheFilter;
+import cgeo.geocaching.ui.TextParam;
+import cgeo.geocaching.ui.ViewUtils;
+
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.LinearLayout;
+
+import androidx.annotation.NonNull;
+
+public class DifficultyAndTerrainFilterViewHolder extends BaseFilterViewHolder<DifficultyAndTerrainGeocacheFilter> {
+
+    private IFilterViewHolder diffView = null;
+    private IFilterViewHolder terrainView = null;
```

Comment:
making these two variables of type "IFilterViewHolder<?>" should make the template warning below go away. I hope...

Reply:
Yes, but unfortunately it will then gives an error in setViewFromFilter
![image](https://user-images.githubusercontent.com/70113341/126715241-d307b976-0a1a-40fe-8d1f-b4304fd78ae4.png)

Reply:
What if you change the type of "diffView" to "IFilterViewHolder<DifficultyGeocacheFilter>"?

I know, generics are tricky...

Reply:
Then I get there the warning of raw use ...

Reply:
I assume you want to get this done for tonights beta? Then maybe ignore the raw use warning for now, I can take a closer look once I have the code to look into (in release)

Reply:
Yes, that was my thought as well, I still have some changes, but my computer is currently not really working, and we have an appointment now.
So probably the remaining stuff is just nice to have and I can clean it later.

So if you think the current state is ok, please feel free to merge, and I try in a few hours again...

If I need to convert from draft, probably that will work from my phone

Reply:
Ah, I think removing the single filter is still local on my machine...